src/lamplib/tasks/download.py

```diff
@@ -0,0 +1,134 @@
+                url = self._get_url()
+                urllib.request.urlretrieve(url, tarball)
+                logging.info('Finished Downloading %s as %s', self._name, tarball)
+
+            logging.info('Extracting %s into %s, please wait...', self._name, self.result_dir)
+
+            shutil.rmtree(self.result_dir, ignore_errors=True)
+            os.mkdir(self.result_dir)
+            # use tar(1) because python's TarFile was inexplicably truncating the tarball
+            subprocess.run(['tar', '-xzf', tarball, '-C', self.result_dir], check=True)
+            logging.info('Finished extracting %s into %s', self._name, self.result_dir)
+
+            # Get space back.
+            os.remove(tarball)
+
+        return True
+
+
+    def _get_url(self):
+        raise NotImplementedError
+
+    def _can_ignore(self):
+        raise NotImplementedError
+
+class ToolchainDownloader(Downloader):
+    TOOLCHAIN_BUILD_ID = 'cd5a4031b1dc93e47d598ad41521fd2e8aa865a0_20_02_12_20_07_30'
+    TOOLCHAIN_GIT_HASH = TOOLCHAIN_BUILD_ID.split('_')[0]
+    TOOLCHAIN_ROOT = '/data/mci'  # TODO BUILD-7624 change this to /opt.
+
+    def __init__(self, os_family, distro):
+        super().__init__(os_family, distro, ToolchainDownloader.TOOLCHAIN_ROOT, "gennytoolchain")
+
+    def _get_url(self):
+        # These build IDs are from the genny-toolchain Evergreen task.
+        # https://evergreen.mongodb.com/waterfall/genny-toolchain
+        if self._os_family == 'Darwin':
+            self._distro = 'macos_1014'
+
+        return 'https://s3.amazonaws.com/mciuploads/genny-toolchain/' \
+               'genny_toolchain_{}_{}/gennytoolchain.tgz'.format(self._distro, ToolchainDownloader.TOOLCHAIN_BUILD_ID)
+
+    def _can_ignore(self):
+        # If the toolchain dir is outdated or we ignore the toolchain version.
+        return os.path.exists(self.result_dir) and (Context.IGNORE_TOOLCHAIN_VERSION or self._check_toolchain_githash())
+
+    def _check_toolchain_githash(self):
+        res = subprocess.run(['git', 'rev-parse', 'HEAD'], cwd=self.result_dir, capture_output=True, text=True)
+        return res.stdout.strip() == ToolchainDownloader.TOOLCHAIN_GIT_HASH
+
+
+class CuratorDownloader(Downloader):
+    CURATOR_VERSION = '9fee6c2020c3d85bbe8fffa03e1d0e224c9652f5'
+    CURATOR_BUILD_DATE = '20_02_12_21_20_42'
+    CURATOR_ROOT = os.getcwd()
+
+    def __init__(self, os_family, distro):
+        super().__init__(os_family, distro, CuratorDownloader.CURATOR_ROOT, "curator")
+
+    def _get_url(self):
+        # These build IDs are from the Curator Evergreen task.
+        # https://evergreen.mongodb.com/waterfall/genny-toolchain
+        if self._os_family == 'Darwin':
+            self._distro = 'macos'
+
+        if 'ubuntu' in self._distro:
+            self._distro = 'ubuntu1604'
+
+        if self._distro in ('amazon2', 'rhel8', 'rhel62'):
+            self._distro = 'rhel70'
+
+        return 'https://s3.amazonaws.com/mciuploads/curator/' \
```

Comment:
Forgot to mention that you should get curator from the push task, that way you have a shorter URL. E.g. https://s3.amazonaws.com/boxes.10gen.com/build/curator/curator-dist-rhel70-6f7afe441326c1a6fa726e341d092839369b9d95.tar.gz

This way you won't need to record the build date